src/index.test.ts

```diff
@@ -184,4 +189,23 @@ describe("scanXmlReport()", () => {
     const severity = "Warning"
     expect(messageCallback).toBeCalledWith(msg, file, line, severity)
   })
+
+  it("returns correct file location when root is different", async () => {
+    const git = {
+      modified_files: ["feature/src/main/res/layout/fragment_password_reset.xml"],
+      created_files: [],
+    }
+
+    mockFiles.push("/otherRoot/feature/src/main/res/layout/fragment_password_reset.xml")
```

Comment:
Instead of maintaining a list of mocked files, why don't you mock the implementation of `mockFileExistsSync` using a `jest.fn` and `.mockImplementation` call?

Reply:
I could do that if you think that is better. I guess what you are suggesting is instead of keeping `mockFiles` around, I'd keep a variable for `mockExistsSync` that will be a `jest.fn()`, and then in the test do something like:

```
mockExistsSync.mockImplementation( path => path === "/otherRoot/feature/src/main/res/layout/fragment_password_reset.xml" )
```

I guess keeping the list of files around makes it easier to read when you have multiple files, or if you wanted to keep the state of which files are present or not across tests.

Reply:
You already have the variable `mockFileExistsSync` available, so it would be one variable less even.

I think using a `jest.fn` + `mockXXX` is more consistent with jest overall, because there are utility functions regarding the mock implementation (reset, restore etc).

I don't think it would be good to have mock data be defined by one test and used in the next, because then these tests always have to run in sequence and it makes the tests harder to understand IMO.

You can still easily handle multiple files in the mock e.g.

```ts
mockExistsSync.mockImplementation((path) => ['file1', 'file2'].includes(path))
```

Reply:
Makes sense. Will do the changes. Thanks!